### PR TITLE
console: persist page size in local storage

### PR DESCRIPTION
**Pull request by beerose**
### Description

This PR makes page size persistent by storing it in local storage.

### Changelog

- [x] `CHANGELOG.md` is updated with user-facing content relevant to this PR.

### Affected components

- [x] Console

### Related Issues

https://github.com/hasura/graphql-engine/issues/3849

### Solution and Design

Added key `data:pageSize` to LS.

### Ideas

A good starting point to refactor local storage utils?

### Steps to test and verify

Go to `data` -> `table`. Change number of rows. Refresh page. The number of rows should be the same.

### CODE REVIEW

**Comment by beerose** [comment]
Makes sense. Will update the PR.

**Comment by rikinsk** [comment]
@beerose Something seems to have gone wrong while setting the limit I think. Getting an error in the fetch rows query  See: https://hge-ci-pull-4323.herokuapp.com/console/data/schema/public/tables/test1/browse

**Comment by beerose** [comment]
Fixed. Since I also changed how page size is stored in LS, you may need to clear local storage.

File: console/src/components/Services/Data/TableBrowseRows/FilterActions.js [hunk 1]
```diff
@@ -133,7 +133,8 @@ const filterReducer = (state = defaultCurFilter, action) => {
           'order_by' in q
             ? [...q.order_by, ...defaultCurFilter.order_by]
             : [...defaultCurFilter.order_by];
-        newCurFilterQ.limit = 'limit' in q ? q.limit : defaultCurFilter.limit;
+        newCurFilterQ.limit =
+          'limit' in q && q.limit ? q.limit : defaultCurFilter.limit;
```
Comment (wawhal) [inline console/src/components/Services/Data/TableBrowseRows/FilterActions.js:137]: Wouldn't this be the same thing?

```suggestion
           q.limit || defaultCurFilter.limit;
```

Comment (wawhal) [reply console/src/components/Services/Data/TableBrowseRows/FilterActions.js:137]: Same for `offset` below this.

File: console/src/components/Services/Data/TableBrowseRows/localStorageUtils.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
+const setLSState = (key: string, data: any) => {
+  window.localStorage.setItem(key, JSON.stringify(data));
+};
+
+const getLSState = (key: string, defaultValue: null | {} = {}) => {
```
Comment (wawhal) [inline console/src/components/Services/Data/TableBrowseRows/localStorageUtils.ts:5]: Does this assume only JSON type values?
In that case, we must call the function something else. Maybe `getJSONFromLS` or something? Same for `setLSState`.

I'd rather prefer `getLSState` function that just returns a string and another util function that takes a string and parses it into whatever you want it to parse as.

File: console/src/components/Services/Data/TableBrowseRows/localStorageUtils.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+const columnsOrderKey = 'data:order';
+
+type OrderEntry = { newOrder: number; defaultOrder: number };
+type OrderState = Record<string, OrderEntry[]>;
+
+const compareReorderItems = (item1: OrderEntry) => (item2: OrderEntry) =>
+  item1.newOrder === item2.newOrder &&
+  item1.defaultOrder === item2.defaultOrder;
+
+export const handleOrderChange = (
```
Comment (wawhal) [inline console/src/components/Services/Data/TableBrowseRows/localStorageUtils.ts:52]:
```suggestion
export const persistColumnOrderChange = (
```

File: console/src/components/Services/Data/TableBrowseRows/localStorageUtils.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    return defaultValue;
+  }
+};
+
+const columnsCollapsedKey = 'data:collapsed';
+
+type CollapesEntry = Record<string, boolean>;
+type CollapesState = Record<string, CollapesEntry>;
+
+export const handleCollapseChange = (
```
Comment (wawhal) [inline console/src/components/Services/Data/TableBrowseRows/localStorageUtils.ts:24]:
```suggestion
export const persistColumnCollapseChange = (
```

File: console/src/components/Services/Data/TableBrowseRows/localStorageUtils.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  const currentCollapsed = getLSState(columnsCollapsedKey);
+  const newCollapsed = {
+    ...currentCollapsed,
+    [`${schemaName}.${tableName}`]: collapsedData,
+  };
+
+  setLSState(columnsCollapsedKey, newCollapsed);
+};
+
+export const getCollapsedColumns = (tableName: string, schemaName: string) => {
```
Comment (wawhal) [inline console/src/components/Services/Data/TableBrowseRows/localStorageUtils.ts:38]:
```suggestion
export const getPersistedCollapsedColumns = (tableName: string, schemaName: string) => {
```

File: console/src/components/Services/Data/TableBrowseRows/localStorageUtils.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+};
+
+export const getColumnsOrder = (tableName: string, schemaName: string) => {
+  const orderData = getLSState(columnsOrderKey) as OrderState;
+  return orderData[`${schemaName}.${tableName}`];
+};
+
+const pageSizeKey = 'data:pageSize';
+
+export const handlePageSizeStateChange = (pageSize: number) => {
```
Comment (wawhal) [inline console/src/components/Services/Data/TableBrowseRows/localStorageUtils.ts:89]:
```suggestion
export const persistPageSizeChange = (pageSize: number) => {
```

**Pull request merged**